test/integration/provider/admin/applications/applications_controller_test.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Provider::Admin::ApplicationsTest < ActionDispatch::IntegrationTest
+  class MasterLoggedInTest < Provider::Admin::ApplicationsTest
+    attr_reader :service
+
+    setup do
+      login! master_account
+      @service = master_account.default_service
+      FactoryBot.create(:cinstance, service: service)
+    end
+
+    test 'index retrieves all master\'s provided cinstances except those whose buyer is master' do
+      get provider_admin_applications_path
+
+      assert_response :ok
+      expected_cinstances_ids = master_account.provided_cinstances.not_bought_by(master_account).pluck(:id)
+      assert_same_elements expected_cinstances_ids, assigns(:cinstances).map(&:id)
```

Comment:
`assigns` are deprecated, you probably want to do a functional test but it is also deprecated

So there are few approaches:

- test that you get the needed data in the view
- use service object approach, and test that the service object is called correctly (expect) in the controller

Both can be combined ;)